hashring/checksum.go

```diff
@@ -0,0 +1,67 @@
+package hashring
+
+import (
+	"bytes"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/dgryski/go-farm"
+)
+
+// Checksum computes a checksum for an instance of a HashRing. The
+// checksum can be used to compare two rings for equality.
+type Checksum interface {
```

Comment:
I don't think we should have an interface here. A simple `func Compute(ring *HashRing)` function would be sufficient. Any reason we want this interface.

Also the interface name Checksumer with a method would be more idiomatic I think.